### PR TITLE
fix: flaky `peg_outs_are_only_allowed_once_per_epoch`

### PR DESCRIPTION
**Pull request by dpc**
I actually didn't know why it's failing and had to do some debugging
and decided to leave these in.

### CODE REVIEW

File: integrationtests/tests/tests.rs [hunk 1]
```diff
@@ -123,27 +124,37 @@ async fn peg_outs_are_rejected_if_fees_are_too_low() -> Result<()> {
...
+        assert!(received1 == sats(0) || received2 == sats(0));
+
+        assert_eq!(
+            user.total_notes().await,
+            sats(5000 - 2 * 1000) - fees - fees
+        );
         user.client.reissue_pending_notes(rng()).await.unwrap();
-        fed.run_consensus_epochs(2).await; // reissue the notes from the tx that failed
+        fed.await_consensus_epochs(2).await.unwrap(); // reissue the notes from the tx that failed
+        user.client.fetch_all_notes().await.unwrap();
```
Comment (dpc) [inline integrationtests/tests/tests.rs:156]: I don't understand how it worked before without this. After we call `reissue_pending_notes`, we need to fetch the new issuance, no?

@jkitman

File: client/client-lib/src/lib.rs [hunk 2]
```diff
@@ -633,14 +636,19 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
 
     /// Should be called after any transaction that might have failed in order to get any note
     /// inputs back.
+    #[instrument(skip_all, level = "debug")]
```
Comment (dpc) [inline client/client-lib/src/lib.rs:639]: https://github.com/tokio-rs/tracing/issues/2460

File: fedimint-api/src/tiered_multi.rs [hunk 3]
```diff
@@ -49,6 +49,11 @@ impl<T> TieredMulti<T> {
         self.0.keys()
     }
 
+    /// Returns the summary of number of items in each tier
+    pub fn summary(&self) -> Tiered<usize> {
```
Comment (dpc) [inline fedimint-api/src/tiered_multi.rs:53]: I need to turn it into a separate struct that displays counts per tier and total together.

**Review by elsirion — APPROVED** [review]
Looks good to me

**Comment by dpc** [comment]
Merging, but please @jkitman help me!

I pushed us into this deep water of parallel test execution, and turns out I don't always have enough context to fix issues that arise with it. :D

File: client/client-lib/src/lib.rs [hunk 4]
```diff
@@ -654,15 +662,18 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
...
             let (key, notes) = result?;
-            all_notes.extend(notes);
+            notes_to_reissue.extend(notes);
             dbtx.remove_entry(&key).await.expect("DB Error");
         }
         dbtx.commit_tx().await.expect("DB Error");
 
-        self.reissue(all_notes, rng).await
+        debug!(target: LOG_WALLET, notes_to_reissue = ?notes_to_reissue.summary(), total = %notes_to_reissue.total_amount());
+        trace!(target: LOG_WALLET, ?notes_to_reissue, "foo");
```
Comment (jkitman) [inline client/client-lib/src/lib.rs:674]: Accidentally merged?

File: integrationtests/tests/tests.rs (hunk 1 above)
Comment (jkitman) [reply integrationtests/tests/tests.rs:156]: The `fetch_all_notes` is in the assert removed below.

File: client/client-lib/src/lib.rs (hunk 4 above)
Comment (dpc) [reply client/client-lib/src/lib.rs:674]: Nope. I think these are useful, and helped me to debug this.

**Pull request merged**